indico/modules/events/registration/client/js/form_submission/Captcha.jsx

```diff
@@ -0,0 +1,124 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2022 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import generateCaptcha from 'indico-url:event_registration.api_generate_captcha';
+
+import PropTypes from 'prop-types';
+import React, {useEffect, useState} from 'react';
+import {Message, Icon, Button, Form, Popup, Placeholder} from 'semantic-ui-react';
+
+import {FinalInput} from 'indico/react/forms';
+import {Translate} from 'indico/react/i18n';
+import {indicoAxios} from 'indico/utils/axios';
+
+import './Captcha.module.scss';
+
+export default function Captcha({name}) {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
+  const [playing, setPlaying] = useState(false);
+  const [image, setImage] = useState(null);
+  const [audio, setAudio] = useState(null);
+
+  const fetchCaptcha = () => {
+    setLoading(true);
+    indicoAxios
+      .get(generateCaptcha())
+      .then(({data}) => {
+        setImage(`data:image/png;base64,${data.image}`);
+        const newAudio = new Audio(`data:audio/mp3;base64,${data.audio}`);
+        newAudio.onended = () => setPlaying(false);
+        setAudio(newAudio);
+        setError(false);
+      })
+      .catch(() => setError(true))
+      .finally(() => setLoading(false));
+  };
+  useEffect(fetchCaptcha, []);
+
+  const toggleAudio = () => {
+    if (playing) {
+      audio.pause();
```

Comment:
hm... do you really want to pause here or rather stop (so it starts from the beginning when you click play again)?